tasks.py

```diff
@@ -880,58 +880,59 @@ def check_schema(context, api_version=None):
-        "pattern": "Only run tests which match the given substring. Can be used multiple times.",
-        "failfast": "Fail as soon as a single test fails don't run the entire test suite.",
-        "buffer": "Discard output from passing tests.",
-        "tag": "Run only tests with the specified tag. Can be used multiple times.",
-        "exclude_tag": "Do not run tests with the specified tag. Can be used multiple times.",
-        "verbose": "Enable verbose test output.",
-        "append": "Append coverage data to .coverage, otherwise it starts clean each time.",
         "parallel": "Run tests in parallel; auto-detects the number of workers if not specified with `--parallel-workers`.",
         "parallel_workers": "Specify the number of workers to use when running tests in parallel.",
+        "pattern": "Only run tests which match the given substring. Can be used multiple times.",
+        "reusedb": "Reuse previously saved test database for faster re-testing in combination with `--keepdb`.",
         "skip_docs_build": "Skip (re)build of documentation before running the test.",
-        "performance_report": "Generate Performance Testing report in the terminal. Has to set GENERATE_PERFORMANCE_REPORT=True in settings.py",
-        "performance_snapshot": "Generate a new performance testing report to report.yml. Has to set GENERATE_PERFORMANCE_REPORT=True in settings.py",
+        "tag": "Run only tests with the specified tag (e.g. 'unit', 'integration', 'migration_test'). Can be used multiple times.",
+        "verbose": "Enable verbose test output.",
     },
     iterable=["tag", "exclude_tag", "pattern"],
 )
-def unittest(
+def tests(
     context,
+    append_coverage=False,
+    buffer=True,
     cache_test_fixtures=True,
+    coverage=False,
+    exclude_tag=None,
+    failfast=False,
     keepdb=True,
-    reusedb=True,
     label="nautobot",
-    pattern=None,
-    failfast=False,
-    buffer=True,
-    exclude_tag=None,
-    tag=None,
-    verbose=False,
-    append=False,
     parallel=True,
     parallel_workers=None,
+    pattern=None,
+    reusedb=True,
     skip_docs_build=False,
-    performance_report=False,
-    performance_snapshot=False,
+    tag=None,
+    verbose=False,
 ):
-    """Run Nautobot unit tests."""
+    """Run Nautobot automated tests."""
     if not skip_docs_build:
         # First build the docs so they are available.
         build_and_check_docs(context)
 
-    if not append:
+    if coverage and not append_coverage:
         run_command(context, "coverage erase")
 
     if parallel_workers:
         parallel_workers = int(parallel_workers)
 
-    append_arg = " --append" if append and not parallel else ""
-    parallel_arg = " --parallel-mode" if parallel else ""
-    command = f"coverage run{append_arg}{parallel_arg} --module nautobot.core.cli test {label}"
+    if coverage:
+        append_arg = " --append" if append_coverage and not parallel else ""
```

Comment:
Why can we not append coverage in parallel mode?

Reply:
In `parallel-mode` coverage creates a separate `.coverage` file for each parallel process, which we then combine later with `coverage join`. These files are randomly named so IIRC we don't need to ever append to an existing file in that case, just auto-create new files on each rerun.